pcapy.cc

```diff
@@ -319,3 +341,268 @@ initpcapy(void)
+            return FALSE; /* Not a hex digit */
+        }
+        guid_text++;
+        digit2 = gethexdigit(guid_text);
+        if(digit2 == -1){
+            return FALSE; /* Not a hex digit */
+        }
+        guid_text++;
+        guid->Data4[i] = (digit1 << 4)|(digit2);
+    }
+    /* Now there must be a hyphen */
+    if(*guid_text != '-'){
+        return FALSE; /* Nope */
+    }
+    guid_text++;
+    /*
+     * There must be 12 hex digits; if so,t hey go into the next 6 bytes
+     * of guid->Data4.
+     */
+    for(i = 0; i < 6; i++){
+        digit1 = gethexdigit(guid_text);
+        if(digit1 == -1){
+            return FALSE; /* Not a hex digit */
+        }
+        guid_text++;
+        digit2 = gethexdigit(guid_text);
+        if(digit2 == -1){
+            return FALSE; /* Not a hex digit */
+        }
+        guid_text++;
+        guid->Data4[i+2] = (digit1 << 4)|(digit2);
+    }
+    /* Now there must be a closing } */
+    if(*guid_text != '}'){
+        return FALSE; /* Nope */
+    }
+    guid_text++;
+    /* And that must be the end of the string */
+    if(*guid_text != '\0'){
+        return FALSE; /* Nope */
+    }
+    return TRUE;
+}
+
+
+ static char* luid_to_guid(char *luid)
+{
+  char errbuff[PCAP_ERRBUF_SIZE];
+  pcap_if_t *devs;
+
+   int status = pcap_findalldevs(&devs, errbuff);
+  if(status)
+    {
+      PyErr_SetString(PcapError, errbuff);
+      return NULL;
+    }
+
+   if(devs==NULL)
+    {
+      PyErr_SetString(PcapError, "No valid interfaces to open");
+      return NULL;
+    }
+
+   pcap_if_t *cursor = devs;
+
+     char* cursor_luid;
+    char *ret_guid = NULL;
+    while(cursor)
+    {
+        cursor_luid = get_windows_interface_friendly_name(cursor->name);
+
+
+         if (!strcmp(cursor_luid, luid))
+        {
+            ret_guid = strdup(cursor->name);
+            goto done;
+        }
+
+         free(cursor_luid);
+
+         cursor = cursor->next;
+    }
+
+ done:
+
+   pcap_freealldevs(devs);
+
+   return ret_guid;
+}
+
+ static char* guid_to_luid(GUID *guid)
+{
+    BOOL status;
+    int size;
+    char *name;
+
+     WCHAR wName[128 + 1];
+    HMODULE hIPHlpApi = LoadLibrary(TEXT("iphlpapi.dll"));
+
+     typedef HRESULT (WINAPI *ProcAddr_nhGINFG) ( GUID *InterfaceGuid,  PCWSTR InterfaceAlias, DWORD *LengthAddress, wchar_t *a4, wchar_t *a5);
+
+     ProcAddr_nhGINFG Proc_nhGetInterfaceNameFromGuid = NULL;
+    Proc_nhGetInterfaceNameFromGuid = (ProcAddr_nhGINFG) GetProcAddress(hIPHlpApi, "NhGetInterfaceNameFromGuid");
+    if (Proc_nhGetInterfaceNameFromGuid!= NULL)
+    {
+        wchar_t *p4=NULL, *p5=NULL;
+        DWORD NameSize;
+
+         /* testing of nhGetInterfaceNameFromGuid indicates the unpublished API function expects the 3rd parameter
+        * to be the available space in bytes (as compared to wchar's) available in the second parameter buffer
+        * to receive the friendly name (in unicode format) including the space for the nul termination.*/
```

Comment:
Thats some very scary stuff :smile:

Not only does this PR breaks backward compatibility, and uses a name that **may suffer from duplication** (yes, you can change an adapter's name), but it also uses an undeclared function, that could disappear at anytime.

I know the intent is good, but there is a reason to use very unfriendly GUIDs: they are unique !